deploy/sdk/src/dynamo/sdk/request_tracing.py

```
@@ -0,0 +1,341 @@
+
+        if "request" in kwargs and isinstance(kwargs["request"], Request):
+            request = kwargs["request"]
+
+        if request:
+            headers_opt = request.headers if hasattr(request, "headers") else None
+            request_id = extract_or_generate_request_id(headers_opt)
+            # preserve previous value (if any) for nested invocations
+            _token = _request_id_ctx.set(request_id)
+            logger.debug(
+                f"Request tracing: extracted/generated request_id={request_id}"
+            )
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)